app/services/flow/flow_state_machine.rb

```diff
@@ -142,25 +142,29 @@ def redirect_to_step(step)
 
     def analytics_submitted
       @analytics_id + ' submitted'
+      "#{@analytics_id} submitted: #{current_step_name}"
     end
 
     def analytics_visited
-      @analytics_id + ' visited'
+      "#{@analytics_id} visited: #{current_step_name}"
     end
 
     def analytics_optional_step
       [@analytics_id, 'optional submitted'].join(' ')
     end
 
     def analytics_properties
-      current_step_name = "#{current_step.to_s}_#{action_name}"
       current_flow_step_counts[current_step_name] ||= 0
       {
         step: current_step,
         step_count: current_flow_step_counts[current_step_name] += 1,
       }
     end
 
+    def current_step_name
+      "#{current_step.to_s}_#{action_name}"
```

Comment:
supernit: `.to_s` inside of `#{}` blocks is redunant, whatever is inside them gets `.to_s` called on it anyways

```suggestion
      "#{current_step}_#{action_name}"
```